packages/compass-sidebar/src/components/multiple-connections/active-connections/active-connection-navigation.tsx

```diff
@@ -499,58 +515,88 @@ export function ActiveConnectionNavigation({
         case 'connection-performance-metrics':
           openPerformanceWorkspace(connectionId);
           return;
-        case 'select-database':
-          openCollectionsWorkspace(connectionId, ns);
-          return;
-        case 'select-collection':
-          openCollectionWorkspace(connectionId, ns);
-          return;
-        case 'open-in-new-tab':
-          openCollectionWorkspace(connectionId, ns, { newTab: true });
-          return;
-        case 'modify-view': {
-          const coll = findCollection(
-            ns,
-            connections.find((conn) => conn.connectionInfo.id === connectionId)
-              ?.databases ?? []
-          );
-          if (coll && coll.sourceName && coll.pipeline) {
-            openEditViewWorkspace(connectionId, coll._id, {
-              sourceName: coll.sourceName,
-              sourcePipeline: coll.pipeline,
-              newTab: true,
-            });
+      }
+    },
+    [
+      onDisconnect,
+      onOpenConnectionInfo,
+      onCopyConnectionString,
+      onToggleFavoriteConnection,
+      openPerformanceWorkspace,
+      openDatabasesWorkspace,
+      openShellWorkspace,
+    ]
+  );
+
+  const onItemAction = useCallback(
+    (item: SidebarActionableItem, action: Actions) => {
+      if (item.type === 'connection') {
+        onConnectionAction(item.connectionInfo.id, action);
+        return;
+      } else {
+        const { connectionId } = item;
+        const ns = item.type === 'database' ? item.dbName : item.namespace;
+        switch (action) {
+          case 'select-database':
+            openCollectionsWorkspace(connectionId, ns);
+            return;
+          case 'select-collection':
+            openCollectionWorkspace(connectionId, ns);
+            return;
+          case 'open-in-new-tab':
+            openCollectionWorkspace(connectionId, ns, { newTab: true });
+            return;
+          case 'modify-view': {
+            const coll = findCollection(
+              ns,
+              (connections.find(
+                (conn) => conn.connectionInfo.id === connectionId
+              )?.databases as Database[]) ?? []
```

Comment:
something's off here, the typecasting shouldn't be necessary 🤔

Reply:
i will check this one

Reply:
this is gone now with [670bbf4](https://github.com/mongodb-js/compass/pull/5886/commits/670bbf42cdb538e599f17697cdf67473e7ec230c)